python/swiftnav/pvt.pyx

```diff
@@ -7,73 +7,53 @@
-
-from common cimport *
-from track_c cimport navigation_measurement_t
+from libc.string cimport memset
 from track cimport NavigationMeasurement
+from track cimport navigation_measurement_t
 
-cdef class Solution:
-  def __repr__(self):
-    return "<Solution %f, %f, %f>" % (57.2957795*self.pos_llh[0],
-                                      57.2957795*self.pos_llh[1],
-                                      self.pos_llh[2])
+cdef class GNSSSolution:
 
-  property pos_llh:
-    def __get__(self):
-      return (self.soln.pos_llh[0],
-              self.soln.pos_llh[1],
-              self.soln.pos_llh[2])
+  def __init__(self, **kwargs):
+    memset(&self._thisptr, 0, sizeof(gnss_solution))
+    if kwargs:
+      self._thisptr = kwargs
 
-  property pos_ecef:
-    def __get__(self):
-      return (self.soln.pos_ecef[0],
-              self.soln.pos_ecef[1],
-              self.soln.pos_ecef[2])
+  def __getattr__(self, k):
+    return self._thisptr.get(k)
 
-  property vel_ned:
-    def __get__(self):
-      return (self.soln.vel_ned[0],
-              self.soln.vel_ned[1],
-              self.soln.vel_ned[2])
 
-  property vel_ecef:
-    def __get__(self):
-      return (self.soln.vel_ecef[0],
-              self.soln.vel_ecef[1],
-              self.soln.vel_ecef[2])
+cdef class DOPS:
 
-  property tow:
-    def __get__(self):
-      return self.soln.time.tow
+  def __init__(self, **kwargs):
+    memset(&self._thisptr, 0, sizeof(dops_t))
+    if kwargs:
+      self._thisptr = kwargs
 
-  property clock_offset:
-    def __get__(self):
-      return self.soln.clock_offset
+  def __getattr__(self, k):
+    return self._thisptr.get(k)
 
-  property clock_bias:
-    def __get__(self):
-      return self.soln.clock_bias
+def calc_PVT_(nav_meas, disable_raim=False):
+  """Wraps the function :libswiftnav:`calc_PVT`.
 
-  property dops:
-    def __get__(self):
-      return (self.dops.pdop,
-              self.dops.gdop,
-              self.dops.tdop,
-              self.dops.hdop,
-              self.dops.vdop)
+  Parameters
+  ----------
+  nav_meas : [NavigationMeasurement]
+    Navigation measurements
+  disable_raim : bool
+    Disable RAIM
 
-def calc_PVT(nav_meas):
-  n_used = len(nav_meas)
-  cdef navigation_measurement_t* nav_meas_array = <navigation_measurement_t*>malloc(n_used*sizeof(navigation_measurement_t))
+  Returns
+  -------
+  (GNSSSolution, DOPS)
 
+  """
+  n_used = len(nav_meas)
+  cdef navigation_measurement_t* nav_meas_ = <navigation_measurement_t *>malloc(n_used*sizeof(navigation_measurement_t))
   for n in range(n_used):
-    nav_meas_array[n] = (<NavigationMeasurement?>nav_meas[n]).meas
-
-  s = Solution()
-  pvt_c.calc_PVT(n_used, nav_meas_array, False, &(s.soln), &(s.dops))
-
-  free(nav_meas_array)
-
-  return s
+    nav_meas_[n] = (<NavigationMeasurement?>nav_meas[n])._thisptr
+  dops_ = DOPS()
+  soln = GNSSSolution()
+  cdef u8 ret = calc_PVT(n_used, nav_meas_, disable_raim, &soln._thisptr, &dops_._thisptr)
```

Comment:
I noticed that when I provided invalid inputs to calc_PVT (before this change) it would sometimes fail, but still return solutions with x, y, z values of 0.  Perhaps this would be a good place to check for success and raise and propagate any errors if they occurred?